### PR TITLE
tart push: allow pushing OCI VMs from the cache too

### PR DESCRIPTION
**Pull request by edigaryev**
Resolves https://github.com/cirruslabs/tart/issues/423.

Supersedes https://github.com/cirruslabs/tart/pull/461.

### CODE REVIEW

File: Sources/tart/VMStorageOCI.swift [hunk 1]
```diff
@@ -16,10 +16,20 @@ class VMStorageOCI: PrunableStorage {
     VMDirectory(baseURL: vmURL(name)).initialized
   }
 
+  func digest(_ name: RemoteName) throws -> String {
+    let digest = vmURL(name).resolvingSymlinksInPath().lastPathComponent
+
+    if !digest.starts(with: "sha256:") {
+      throw RuntimeError.OCIStorageError("\(name) is not a digest and doesn't point to a digest")
+    }
```
Comment (fkorotkov) [inline Sources/tart/VMStorageOCI.swift:24]: Does it mean that we'll be able to push only `sha256` digests and won't be able to push tags?

```bash
tart push ghcr.io/cirruslabs/macos-ventura-xcode:14.3-rc-2 ghcr.io/cirruslabs/macos-ventura-xcode:14.3
```

Comment (edigaryev) [reply Sources/tart/VMStorageOCI.swift:24]: The `resolvingSymlinksInPath()` above resolves any tag into a digest, so we'll be able to push both tags and digests.

It's just a sanity check to detect potential errors.

File: Sources/tart/Commands/Push.swift [hunk 2]
```diff
@@ -53,23 +53,63 @@ struct Push: AsyncParsableCommand {
...
         for remoteName in remoteNamesForRegistry {
           try ociStorage.link(from: remoteName, to: pushedRemoteName)
         }
       }
     }
   }
+
+  func lightweightPushToRegistry(registry: Registry, references: [String]) async throws -> RemoteName? {
+    // Is the local name we're pushing is actually an OCI VM?
+    guard let remoteName = try? RemoteName(localName) else {
```
[the remark this answers is not in the document]
Comment (edigaryev) [reply Sources/tart/Commands/Push.swift]: See https://github.com/cirruslabs/tart/pull/465/commits/94a672bbd8913a98cf676105f50ff25eebe563f3.

File: Sources/tart/Commands/Push.swift [hunk 3]
```diff
@@ -53,23 +53,63 @@ struct Push: AsyncParsableCommand {
...
+        pushedRemoteName = remoteName
+        lightweightPush = true
+      } else {
+        pushedRemoteName = try await localVMDir.pushToRegistry(
+          registry: registry,
+          references: references,
+          chunkSizeMb: chunkSize
+        )
+        lightweightPush = false
+      }
```
[the remark this answers is not in the document]
Comment (edigaryev) [reply Sources/tart/Commands/Push.swift]: See https://github.com/cirruslabs/tart/pull/465/commits/94a672bbd8913a98cf676105f50ff25eebe563f3, do you want to move the rest of the `lightweightPushToRegistry()` in the `if` branch?

[the remark this answers is not in the document]
Comment (fkorotkov) [reply Sources/tart/Commands/Push.swift]: I was thinking in the lines of #466

**Review by fkorotkov — APPROVED** [review]

**Pull request merged**